CMakeLists.txt

```diff
@@ -396,6 +397,13 @@ if (MFEM_USE_AMGX)
   find_package(AMGX REQUIRED)
 endif()
 
+if (MFEM_USE_MAGMA)
+  # MAGMA depends on BLAS and LAPACK
+  find_package(BLAS REQUIRED)
+  find_package(LAPACK REQUIRED)
+  find_package(MAGMA REQUIRED)
```

Comment:
For consistency with other packages, the dependency on `BLAS` and `LAPACK` should be expressed using the variable `MAGMA_REQUIRED_PACKAGES` which should be defined in `defaults.cmake`, see for example how this is done for `SuperLUDist` in `defaults.cmake`.

Also, we should add a default setting for `MAGMA_DIR` in `defaults.cmake`, similar to other packages.

Also, in `FindMAGMA.cmake`, the GPU library dependencies for CUDA or ROCm can be set after auto-detecting if the build uses CUDA or ROCm, similar to how we do that for HYPRE in `FindHYPRE.cmake`.

Reply:
I attempted some of these changes in bc729465d6117a4b9a1bf5235ee4239f3d021c1c.

Once BLAS and LAPACK are added to `MAGMA_REQUIRED_PACKAGES`, should these `find_package` statements be removed?

I made some changes to `FindMAGMA.cmake`, but probably didn't do it correctly.

Reply:
> Once BLAS and LAPACK are added to `MAGMA_REQUIRED_PACKAGES`, should these `find_package` statements be removed?

Yes.

Regarding the other changes: I'll take a look at them and leave comments, if I see something.

Reply:
Reminder: remove `find_package` for `BLAS` and `LAPACK`.

Reply:
Thanks, done in cbbfaede8.